### PR TITLE
Home Highlighted Events

### LINKED ISSUE #16: Home Highlighted Events
Showcase featured highlights in the home page.

| What we're adding | Where we're adding it |
| --- | --- |
| image | image |

### PR DESCRIPTION
**Pull request by phillytan**
**Changelogs:**
- Add Featured Event + Next Event in Homepage
- Fixed call on getting a user's registered events

Closes #16 

State 1 | State 2 | State 3
 --- | --- | ---
simulator_screenshot_123C352C-9ADE-4B7B-B400-D15F85488B50 | simulator_screenshot_123C352C-9ADE-4B7B-B400-D15F85488B50 | simulator_screenshot_123C352C-9ADE-4B7B-B400-D15F85488B50

### CODE REVIEW

File: lib/home/home.dart [hunk 1]
```diff
@@ -17,13 +20,69 @@ class Home extends StatefulWidget {
...
+                alignment: Alignment.center,
+                child: Container(
+                  height: 60,
+                  width: 1.0,
+                  color: C.darkBackgroundCard,
+                ),
+              ),
+            ),
+            if (model.nextEvent != null) FeaturedEvent(
+              headline: 'Your Next Event',
```
Comment (adindotdev) [inline lib/home/home.dart:44]: Should be in the constants file.

File: lib/home/home_presenter.dart [hunk 2]
```diff
@@ -108,6 +114,44 @@ class HomePresenter extends Presenter<HomeView, HomeModel> {
     }
   }
 
+  Future<void> setFeaturedEvent() async {
+    await _eventsManager.loadEvents();
```
Comment (adindotdev) [inline lib/home/home_presenter.dart:118]: Called also on line 130 -> look to abstract?

File: lib/common/user.dart [hunk 3]
```diff
@@ -60,11 +60,12 @@ class User {
     // TODO(anyone): make sure this works, can't test until registrations work.
     SharedPreferences preferences = await SharedPreferences.getInstance();
     try {
-      List<String> registeredEvents = await Fetcher().fetchBackend(
-              '/registrations?id=$studentId', FetcherMethod.get) ??
-          [];
-      preferences.setStringList(_registeredEventsKey, registeredEvents);
-      registeredEventsId = registeredEvents.toSet();
+      Map<String, dynamic> responseData = await Fetcher().fetchBackend(
```
Comment (adindotdev) [inline lib/common/user.dart:63]: Ambiguous variable name.

**Review by adindotdev — CHANGES_REQUESTED** [review]

**Comment by jindazhao** [comment]
lib/events/events_presenter.dart has extra space not letting check pass please fix

**Comment by voctory** [comment]
Closing stale PRs in `ubc-biztech` org

**Pull request closed without merging**